### PR TITLE
build-support/rust: specify proper path to Cargo.lock

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

When building a crate from a subdir (specified by `sourceRoot`), the
diffing fails as the builder tries to compare `source/Cargo.lock` to the
one in `$cargoDepsCopy`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by bhipple — CHANGES_REQUESTED** [review]
This is fixed on the `staging` branch, and will be good to go once merged to `master`.  See here for details:
https://github.com/NixOS/nixpkgs/pull/80262

I've cherry-picked your first commit onto a local checkout of `staging` and verified it builds with no other changes required, so I guess there are three choices:

A) Send the PR bumping `cargo-raze` to staging, or
B) Send the PR to master, but keep using the legacy fetcher until the bugfix makes its way to master, or
C) Wait for staging to merge to master, then send the PR bumping `cargo-raze`

I'm OK with any of them, and if you keep using the legacy fetcher I'll clean it up once staging merges to master.

**Comment by Ma27** [comment]
I see. I guess I should've looked at staging first before filing that patch, sorry! Will push the cargo-raze update as soon as all changes are on master.

**Pull request closed without merging**
